components/chart/index.tsx

```diff
@@ -56,11 +57,36 @@ function Chart({
     ZOOM_IN: () => changeZoomLevel(-10),
     ZOOM_OUT: () => changeZoomLevel(10),
   };
+  useEffect(() => {
+    const queryParams = parse(window.location.search);
+    const nValue = queryParams['n'];
+    if (nValue) {
+      const nodeId = parseInt(nValue as string);
+      setSelectedNodeID(nodeId);
+    } else {
+      setSelectedNodeID(null);
+    }
+  }, []);
+  useEffect(() => {
+    if (selectedNodeID !== null) {
+      const node = findModuleById(selectedNodeID);
+      selectModule(node);
```

Comment:
In case you used setSelectedModule, no need for another useEffect, in the first one you can just find the module and set it

Reply:
done